Sources/Commandant/OrderedSet.swift

```diff
@@ -0,0 +1,49 @@
+/// A poor man's ordered set.
+internal struct OrderedSet<T: Hashable> {
+	fileprivate var values: [T] = []
+
+	init<S: Sequence>(_ sequence: S) where S.Element == T {
+		for e in sequence where !values.contains(e) {
+			values.append(e)
+		}
+	}
+
+	@discardableResult
+	mutating func remove(_ member: T) -> T? {
+		if let index = values.index(of: member) {
+			return values.remove(at: index)
+		} else {
+			return nil
+		}
+	}
+
+	static func == (_ lhs: OrderedSet, rhs: OrderedSet) -> Bool {
```

Comment:
We should declare `OrderedSet` to be `Equatable` then?

Reply:
Absolutely 🙈